packages/react-server/src/ReactFlightServerConfigDebugNode.js

```diff
@@ -81,26 +100,34 @@ export function initAsyncDebugInfo(): void {
               return;
             }
             const current = pendingOperations.get(currentAsyncId);
+            if (current !== undefined && current.promise !== null) {
+              const currentPromise = current.promise.deref();
+              if (currentPromise !== undefined) {
+                // Keep the previous Promise alive as long as the child is alive so we can
+                // trace its value at the end.
+                previousPromise.set(resource, currentPromise);
+              }
+            }
             // If the thing we're waiting on is another Await we still track that sequence
             // so that we can later pick the best stack trace in user space.
             node = ({
               tag: UNRESOLVED_AWAIT_NODE,
               owner: resolveOwner(),
-              debugInfo: new WeakRef((resource: Promise<any>)),
-              stack: parseStackTrace(new Error(), 1),
+              stack: parseStackTrace(new Error(), 5),
```

Comment:
Couldn't the `5` become incorrect when Node.js internals change between versions? Should we ignore `node:internal/async_hooks` frames instead (plus skip the first frame)?